### PR TITLE
cp-277: Group Chat Messages By Date

### PR DESCRIPTION
**Pull request by danilittaa**
- messages are visually separated based on their relative date

image

### CODE REVIEW

**Review by iamAlinaaa — COMMENTED** [review]
I think there should be some more space / padding between grey line of date and message

File: shared/src/libs/helpers/date/get-relative-date/get-relative-date.helper.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { formatRelative } from 'date-fns';
+import enGB from 'date-fns/locale/en-GB/index.js';
+
+import { type ValueOf } from '#libs/types/types.js';
+
+import { FormatRelativeLocale } from '../libs/enums/enums.js';
+
+const getRelativeDate = (date: Date): string => {
```
Comment (v0ldemar01) [inline shared/src/libs/helpers/date/get-relative-date/get-relative-date.helper.ts]: Please check related PR for a web part

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/screens/chat/chat.tsx [hunk 2]
```diff
@@ -49,7 +49,8 @@ const Chat: React.FC = () => {
   );
 
   const scrollViewReference = useRef<ScrollView | null>(null);
-  const messagesLength = currentChatMessages.length;
+  const messagesLength = Object.keys(currentChatMessages).length;
```
Comment (dmytro-kucherenko) [inline mobile/src/screens/chat/chat.tsx]:
```suggestion
  const messagesLength = Object.keys(currentChatMessages).flat().length;
```

I also changed that in my pr, cause before it was only the count of days when messages were sent

File: shared/src/libs/helpers/date/libs/enums/format-relative-locale.enum.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { SINGLE_QUOTE_UNICODE } from '../constants/constants.js';
```
Comment (dmytro-kucherenko) [inline shared/src/libs/helpers/date/libs/enums/format-relative-locale.enum.ts]: The approach in helper was changed, so you can delete this enum at all

File: shared/src/libs/helpers/date/libs/constants/constants.ts [hunk 4]
```diff
@@ -1,4 +1,6 @@
 const MILLISECONDS_IN_SECOND = 1000;
 const SECONDS_IN_MINUTE = 60;
 
-export { MILLISECONDS_IN_SECOND, SECONDS_IN_MINUTE };
+const SINGLE_QUOTE_UNICODE = '\u0027';
```
Comment (dmytro-kucherenko) [inline shared/src/libs/helpers/date/libs/constants/constants.ts]: The same with that constant

File: mobile/src/screens/chat/chat.tsx (hunk 2 above)
Comment (danilittaa) [reply mobile/src/screens/chat/chat.tsx]: Thank you! I wasnt attentive

**Review by liza-veis — APPROVED** [review]

**Review by v0ldemar01 — APPROVED** [review]

**Pull request merged**